docs/docusaurus/docs/core/connect_to_data/sql_data/_create_a_batch_definition/_create_a_batch_definition.md

```diff
@@ -0,0 +1,78 @@
+import TabItem from '@theme/TabItem';
+import Tabs from '@theme/Tabs';
+import GxData from '../../../_core_components/_data.jsx'
+import PreReqDataContext from '../../../_core_components/prerequisites/_preconfigured_data_context.md'
+
+### Prerequisites
+- <PreReqDataContext/>.  The variable `context` is used for your Data Context in the following example code.
+- [A Data Asset on a SQL Data Source](#create-a-data-asset).
+
+<Tabs>
+
+<TabItem value="procedure" label="Procedure">
+
+1. Retrieve your Data Asset.
+
+   Replace the value of `datasource_name` with the name of your Data Source and the value of `asset_name` with the name of your Data Asset in the following code.  Then execute it to retrieve an existing Data Source and Data Asset from your Data Context:
+
+   ```python title="Python" name="docs/docusaurus/docs/core/connect_to_data/sql_data/_create_a_data_asset/create_a_data_asset.py retrieve a Data Asset"
+   ```
+
+2. Add a Batch Definition to the Data Asset.
+
+   <Tabs queryString="batch_definition" groupId="batch_definition" defaultValue='full_table' values={[{label: 'Full table', value:'full_table'}, {label: 'Partitioned', value:'partitioned'}]}>
+
+   <TabItem value="full_table" label="Full table">
+   
+   A full table Batch Definition returns all of the records in your Data Asset as a single Batch.  Therefore, to define a full table Batch Definition you only need to provide a name for the Batch Definition to be referenced by.
+ 
+   Update the `name` parameter and execute the following code to create a full table Batch Definition:
+
+   ```python title="Python" name="docs/docusaurus/docs/core/connect_to_data/sql_data/_create_a_batch_definition/_create_a_batch_definition.md full table batch definition"
+   ```
+   </TabItem>
+
+   <TabItem value="partitioned" label="Partitioned">
+   
+   A partitioned Batch Definition subdivides the records in a Data Asset based on the values in a specified field.  {GxData.product_name} currently supports partitioning Data Assets based on date fields.  The records can be grouped by year, month, or day.
+
+   Update the `date_column` variable and `name` parameters in the following snippet, then execute it to create partitioned Batch Definitions:
+
+   ```python name="docs/docusaurus/docs/core/connect_to_data/sql_data/_create_a_batch_definition/_create_a_batch_definition.md daily batch definition"
+   ```
+   </TabItem>
+
+   </Tabs>
+
+5. Optional. Verify the Batch Definition is valid.
+
+   When retrieving a Batch from a partitioned Batch Definition, you can specify the date of the data to retrieve as shown in the following examples.  If you do not specify a date, the most recent date in the data is returned by default.
+
+   <Tabs queryString="batch_definition" groupId="batch_definition" defaultValue='full_table' values={[{label: 'Full table', value:'full_table'}, {label: 'Partitioned', value:'partitioned'}]}>
+
+   <TabItem value="full_table" label="Full table">
```

Comment:
A couple thoughts here:

`When retrieving a Batch from a partitioned Batch Definition` <- can we make this more general? The same concepts follow for running validation definitions or checkpoints. And these concepts also apply to non-sql datasources. Is there a way to make this general enough that folks can get the feel for how batch definitions and batch parameters work together holistically? I'm imagining something like a table of the different batch definition types that shows what batch parameters are valid to pass to them, and then saying you can pass those batch params when getting a batch, running a checkpoint, or a validation definition.

The part about using `head` to verify that it works seems like it doesn't show the user why it works. Something "like `batch.head()` gives the first few rows; we can use this to verify our batch definition is configured correctly. You will see an error when calling `head()` if the batch def isn't set up properly" or something like that.

kinda similar to other thoughts here, but showing an example of calling `head()` for each batch definition feels to me like it shows each of these batch definitions as being their own thing, when they are really all the same concept and can be used interchangeably. I'd probably just give one general example where you show `batch = batch_definition.get_batch(..)` and `batch.head()`

Reply:
I'm putting in a ticket to come back and review/revise this once more of the GX 1.0 docs are done.  I'd like to simplify some of these guides further, but I don't want to burn a lot of time working out the best way to do that while we still have gaps to cover in the 1.0 docs.